evalml/objectives/objective_base.py

```diff
@@ -10,6 +10,8 @@
 class ObjectiveBase(ABC):
     """Base class for all objectives."""
 
+    positive_only = False
```

Comment:
@ParthivNaresh sweet!

Now that I'm looking at this next to the other attributes we've defined for `ObjectiveBase` I have a few thoughts:
* Let's follow a similar pattern to the other `ObjectiveBase` class attributes. Perhaps we don't need to make this `abstract` since the vast majority of objectives won't change it from the default.
* Whatever we settle on let's make sure a docstring appears in the docs both for a) any objectives where we set this to True and b) all other objectives
* Style nit-pick: I think we should make this the last attribute listed in the file here rather than the first, because its less important than the others.

How about this:
```python
@property
@classmethod
def positive_only(cls):
    """If True, this objective is only valid for positive data. Default False."""
    return False
```

and then in standard metrics for RMSLE/MSLE/MAPE:
```python
@property
@classmethod
def positive_only(cls):
    """If True, this objective is only valid for positive data. Default False."""
    return True
```